src/Http/Http.Api/src/AwaitableInfo.cs

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+
+namespace Microsoft.AspNetCore.Http.Api
+{
+    internal readonly struct AwaitableInfo
```

Comment:
This exists in ObjectMethodExecutor, maybe make it internal there and use it from here so we don't need a copy of the code?